libs/orchestrator-ui-components/src/lib/components/Table/Table.tsx

```diff
@@ -1,140 +1,84 @@
-    ) => EuiDataGridControlColumn = ({ id, width, rowCellRender }) => ({
-        id,
-        width,
-        headerCellRender: (props) => null,
-        rowCellRender: ({ rowIndex }) => {
-            const { pageSize, pageIndex } = pagination;
-            const rowIndexOnPage = rowIndex - pageIndex * pageSize;
-
-            const dataRow = data[rowIndexOnPage];
-
-            return rowCellRender(dataRow);
-        },
-    });
-
-    const euiDataGridLeadingControlColumns = leadingControlColumns?.map(
-        controlColumnToEuiDataGridControlColumnMapper,
-    );
-
-    const euiDataGridTrailingControlColumns = trailingControlColumns?.map(
-        controlColumnToEuiDataGridControlColumnMapper,
-    );
-
-    const gridHeightValue =
-        pagination.pageSize * 40 + EUI_DATA_GRID_HEIGHT_OFFSET;
-
-    return (
-        <EuiDataGrid
-            aria-label="Data Grid"
-            columns={initialColumnOrderRef.current}
-            leadingControlColumns={euiDataGridLeadingControlColumns}
-            trailingControlColumns={euiDataGridTrailingControlColumns}
-            height={`${gridHeightValue}px`}
-            gridStyle={GRID_STYLE}
-            columnVisibility={{ visibleColumns, setVisibleColumns }}
-            pagination={pagination}
-            sorting={columnSortToEuiDataGridSorting(
-                dataSorting,
-                updateDataSorting,
-            )}
-            rowCount={pagination.totalRecords}
-            renderCellValue={renderCellValue}
-        />
-    );
-};
+    pagination,
+    isLoading,
+    onCriteriaChange,
+    onDataSort,
+}: TableProps<T>) => (
+    <EuiBasicTable
+        tableCaption="Demo of EuiBasicTable"
+        items={data}
+        rowHeader="firstName"
+        columns={mapTableColumnsToEuiColumns(
+            columns,
+            hiddenColumns,
+            dataSorting,
+            onDataSort,
+        )}
+        pagination={pagination}
+        onChange={onCriteriaChange}
+        loading={isLoading}
+    />
+);
+
+function mapTableColumnsToEuiColumns<T>(
+    columns: TableColumns<T>,
+    hiddenColumns?: Array<keyof T>,
+    dataSorting?: DataSorting<T>,
+    onDataSort?: (columnId: keyof T) => void,
+): EuiBasicTableColumn<T>[] {
+    function isVisibleColumn(columnKey: string) {
+        return !hiddenColumns?.includes(columnKey as keyof T);
+    }
+
+    function mapToEuiColumn(colKey: string): EuiBasicTableColumn<T> {
+        const typedColumnKey = colKey as keyof T;
+        const column = columns[typedColumnKey];
+        const { name } = column;
+
+        const sortDirection =
+            dataSorting?.columnId === colKey
+                ? dataSorting.sortDirection
+                : undefined;
+
+        return {
+            ...column,
+            field: typedColumnKey,
+            name: name && (
+                <TableHeaderCell
+                    sortDirection={sortDirection}
+                    onClick={() => onDataSort?.(typedColumnKey)}
```

Comment:
I'd like to suggest that we start making such optimizations in case we really have performance issues. For now I implemented some handlers to improve readability of the code.
On the nullish coalescing matter, what fallback would you suggest? onClick can only trigger onDataSort when the consumer of the component defines onDataSort. When not used, noting happens.

Reply:
- I must stand my point.
- It's well known that inline function definitions in component render return trigger unnecessary renders.
- Unnecessary render is not related to the value of `onDataSort`, but to the definition of an inline function as such {() => ... } regardless of its implementation.
- Waiting till having performance issues first and letting not optimized implementation live on in the meanwhile is too late to act upon. This should not become a common practice.

- Implicit fallback value seems to be fair enough in this case.